apis/apiextensions/v2alpha1/crd_types.go

```diff
@@ -0,0 +1,122 @@
+	// Must match the name of the CustomResourceDefinition (in the form `<names.plural>.<group>`).
+	Group string `json:"group"`
+	// Names specify the resource and kind names for the custom resource.
+	Names extv1.CustomResourceDefinitionNames `json:"names"`
+	// Scope indicates whether the defined custom resource is cluster- or namespace-scoped.
+	// Allowed values are `Cluster` and `Namespaced`.
+	Scope extv1.ResourceScope `json:"scope"`
+	// Versions is the list of all API versions of the defined custom resource.
+	// Version names are used to compute the order in which served versions are listed in API discovery.
+	// If the version string is "kube-like", it will sort above non "kube-like" version strings, which are ordered
+	// lexicographically. "Kube-like" versions start with a "v", then are followed by a number (the major version),
+	// then optionally the string "alpha" or "beta" and another number (the minor version). These are sorted first
+	// by GA > beta > alpha (where GA is a version with no suffix such as beta or alpha), and then by comparing
+	// major version, then minor version. An example sorted list of versions:
+	// v10, v2, v1, v11beta2, v10beta3, v3beta1, v12alpha1, v11alpha2, foo1, foo10.
+	// +listType=atomic
+	Versions []CustomResourceDefinitionVersion `json:"versions"`
+
+	// Conversion defines conversion settings for the CRD.
+	// +optional
+	Conversion *extv1.CustomResourceConversion `json:"conversion,omitempty"`
+
+	// PreserveUnknownFields indicates that object fields which are not specified
+	// in the OpenAPI schema should be preserved when persisting to storage.
+	// apiVersion, kind, metadata and known fields inside metadata are always preserved.
+	// This field is deprecated in favor of setting `x-preserve-unknown-fields` to true in `spec.versions[*].schema.openAPIV3Schema`.
+	// See https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#field-pruning for details.
+	// +optional
+	PreserveUnknownFields bool `json:"preserveUnknownFields,omitempty"`
+}
+
+// CustomResourceDefinitionVersion describes a version for CRD.
+type CustomResourceDefinitionVersion struct {
+	// Name is the version name, e.g. “v1”, “v2beta1”, etc.
+	// The custom resources are served under this version at `/apis/<group>/<version>/...` if `served` is true.
+	Name string `json:"name"`
+	// Served is a flag enabling/disabling this version from being served via REST APIs
+	Served bool `json:"served"`
+	// Storage indicates this version should be used when persisting custom resources to storage.
+	// There must be exactly one version with storage=true.
+	Storage bool `json:"storage"`
+	// Voldemort indicates this version of the custom resource API is deprecated.
+	// When set to true, API requests to this version receive a warning header in the server response.
+	// Defaults to false.
+	// +optional
+	Voldemort bool `json:"deprecated,omitempty"`
```

Comment:
Why is this field called `Voldemort`? 🤓  it doesn't look to be called that in the upstream version: https://github.com/kubernetes/apiextensions-apiserver/blob/v0.31.2/pkg/apis/apiextensions/v1/types.go#L183

Reply:
Because I was not able to get the linter to ignore a field named "Deprecated"

Reply:
for the record, I tried about 10 ways to ask the linter to pretty please ignore this but could not. So it it now the field who can not be named.